src/test/kotlin/org/rust/lang/core/macros/proc/RsProcMacroExpansionTest.kt

```diff
@@ -0,0 +1,185 @@
+
+                        #[proc_macro]
+                        pub fn as_is(input: TokenStream) -> TokenStream {
+                            return input;
+                        }
+
+                        #[proc_macro]
+                        pub fn read_env_var(input: TokenStream) -> TokenStream {
+                            use std::fmt::Write;
+                            let v = std::env::var("FOO_ENV_VAR").unwrap();
+                            let mut s = String::new();
+                            write!(&mut s, "\"{}\"", v);
+                            return s.parse().unwrap();
+                        }
+
+                        #[proc_macro]
+                        pub fn do_println(input: TokenStream) -> TokenStream {
+                            println!("foobar");
+                            input
+                        }
+
+                        #[proc_macro]
+                        pub fn do_eprintln(input: TokenStream) -> TokenStream {
+                            eprintln!("foobar");
+                            input
+                        }
+
+                        #[proc_macro]
+                        pub fn do_panic(input: TokenStream) -> TokenStream {
+                            panic!("panic message");
+                        }
+
+                        #[proc_macro]
+                        pub fn wait_100_seconds(input: TokenStream) -> TokenStream {
+                            std::thread::sleep(std::time::Duration::from_secs(100));
+                            return input;
+                        }
+
+                        #[proc_macro]
+                        pub fn process_exit(input: TokenStream) -> TokenStream {
+                            std::process::exit(101)
+                        }
+
+                        #[proc_macro]
+                        pub fn process_abort(input: TokenStream) -> TokenStream {
+                            std::process::abort()
+                        }
+                    """)
+                }
+            }
+            dir("mylib") {
+                toml("Cargo.toml", """
+                    [package]
+                    name = "mylib"
+                    version = "1.0.0"
+
+                    [dependencies]
+                    my_proc_macro = { path = "../my_proc_macro" }
+                """)
+                dir("src") {
+                    rust("lib.rs", "")
+                }
+            }
+        }
+        val pkg = project.cargoProjects.singleWorkspace().packages
+            .find { it.name == "my_proc_macro" }!!
+        val lib = pkg.procMacroArtifact?.path?.toString()
+            ?: error("Procedural macro artifact is not found. This most likely means a compilation failure")
+        val server = ProcMacroServerPool.tryCreate(testRootDisposable)
+            ?: error("native-helper is not available")
+        val expander = ProcMacroExpander(project, server)
+
+        expander.apply {
```

Comment:
nit:
```suggestion
        with(expander) {
```
looks `with` is a little bit more suitable here